README.md

```diff
@@ -1416,38 +1416,126 @@ Default is `eth0`.
 
-or
+Example with NFS:
 
-```shell
-config.vm.synced_folder './', '/vagrant', type: '9p', disabled: false, accessmode: "squash", owner: "1000"
+``` ruby
+Vagrant.configure("2") do |config|
+  config.vm.synced_folder "./", "/vagrant"
+end
 ```
 
-or
+**RSync**
 
-```shell
-config.vm.synced_folder './', '/vagrant', type: '9p', disabled: false, accessmode: "mapped", mount: false
+`vagrant-libvirt` supports
+[rsync](https://www.vagrantup.com/docs/synced-folders/rsync) with
+unidirectional synced folders.
+
+Example with rsync:
+
+``` ruby
+Vagrant.configure("2") do |config|
+  config.vm.synced_folder "./", "/vagrant", type: "rsync"
+end
 ```
 
+**9P**
+
+`vagrant-libvirt` supports [VirtFS](http://www.linux-kvm.org/page/VirtFS) ([9p
+or Plan 9](https://en.wikipedia.org/wiki/9P_\(protocol\))) with bidirectional
+synced folders.
+
+Difference between NFS and 9p is explained
+[here](https://unix.stackexchange.com/questions/240281/virtfs-plan-9-vs-nfs-as-tool-for-share-folder-for-virtual-machine).
+
 For 9p shares, a `mount: false` option allows to define synced folders without
 mounting them at boot.
 
-Further documentation on using 9p can be found in [kernel docs](https://www.kernel.org/doc/Documentation/filesystems/9p.txt) and in [QEMU wiki](https://wiki.qemu.org/Documentation/9psetup#Starting_the_Guest_directly). Please do note that 9p depends on support in the guest and not all distros come with the 9p module by default.
+Example for `accessmode: "squash"` with 9p:
 
-**SECURITY NOTE:** for remote Libvirt, nfs synced folders requires a bridged
-public network interface and you must connect to Libvirt via ssh.
+``` ruby
+Vagrant.configure("2") do |config|
+  config.vm.synced_folder "./", "/vagrant", type: "9p", disabled: false, accessmode: "squash", owner: "1000"
+end
+```
+
+Example for `accessmode: "mapped"` with 9p:
+
+``` ruby
+Vagrant.configure("2") do |config|
+  config.vm.synced_folder "./", "/vagrant", type: "9p", disabled: false, accessmode: "mapped", mount: false
+end
+```
+
+Further documentation on using 9p can be found in [kernel
+docs](https://www.kernel.org/doc/Documentation/filesystems/9p.txt) and in
+[QEMU
+wiki](https://wiki.qemu.org/Documentation/9psetup#Starting_the_Guest_directly).
+Please do note that 9p depends on support in the guest and not all distros
+come with the 9p module by default.
+
+**Virtio-fs**
+
+`vagrant-libvirt` supports [Virtio-fs](https://virtio-fs.gitlab.io/) with
+bidirectional synced folders.
+
+For virtiofs shares, a `mount: false` option allows to define synced folders
+without mounting them at boot.
+
+So far, passthrough is the only supported access mode and it requires running
+the virtiofsd daemon as root.
+
+QEMU needs to allocate the backing memory for all the guest RAM as shared
+memory, e.g. [Use file-backed
+memory](https://libvirt.org/kbase/virtiofs.html#host-setup) by enable
+`memory_backing_dir` option in `/etc/libvirt/qemu.conf`:
+
+``` shell
+memory_backing_dir = "/dev/shm"
```

Comment:
Based on my understanding of the docs you've linked to, if we added support for hugepages for memorybacking subsequently, that would potentially allow people using this to not need to perform additional system configuration? e.g. they can skip modifying the file mentioned above.

Not suggesting that there is any need to do that here, just looking to make sure I'm following how it might be possible to simplify the config for users subsequently.

Reply:
Just tried as below:
```
libvirt.memorybacking :hugepages, :mode => 'shared'
```

But failed with below message:
```
==> default: could not attach device because: Call to virDomainAttachDeviceFlags failed: unsupported configuration: virtiofs requires shared memory
Error while attaching new device to domain. Call to virDomainAttachDeviceFlags failed: unsupported configuration: virtiofs requires shared memory
```

Reply:
Also, as https://libvirt.org/kbase/virtiofs.html#host-setup mentioned:

> Use hugepage-backed memory
> 
> Make sure there are enough huge pages allocated for the requested guest memory. For example, for one guest with 2 GiB of RAM backed by 2 MiB hugepages:
> 
> ```
> # virsh allocpages 2M 1024
> ```

Since it related to https://github.com/alvistack/vagrant-libvirt-vagrant-libvirt/tree/master-virtiofs#memory-backing related (i.e. out-of-scope for synced_folder) and additional consideration is required, IMHO using file-backed memory would be more simple and set-it-and-forget-it ;-)

Reply:
Sure as a starting point, I'm thinking more for systems where people may not have root and are running qemu under a session, it is something I just wanted to confirm as being useful for future investigation.

Reply:
Generally I just adduser myself to kvm and libvirt, then running vagrant without sudo, where virtiofs still working ;-)